### ISSUE TITLE
Graphical Error on Nobara 39 KDE Plasma

### ISSUE THREAD

**Issue by pgrwe** [leonardochappuis/sddmsel#1]
## Image
![image](https://github.com/leonardochappuis/sddmsel/assets/89813338/83e1e46a-515a-4c37-be08-0a1576868ca5)
## Current Troubleshooting Status
Strangely my config file in /etc/sddm.conf was empty... Completely new to linux so I probably messed something up on my end but just submitting in case its a bug. Could also be a wayland + nvidia issue as I have been having trouble there as well.

- Dependencies are installed
- Theme shows up properly in KDE SDDM system settings
- I am able to switch between breeze and nobara themes successfully (gonna try another theme later to test)
- I can hear the startup audio and login normally

![image](https://github.com/leonardochappuis/sddmsel/assets/89813338/b44b1149-eabb-43b1-80fe-41a2fb40c6de)

- Same issue appeared in test mode
- Session info: 
![image](https://github.com/leonardochappuis/sddmsel/assets/89813338/7f3ad00e-af50-4204-95e0-babc210e6e22)
![image](https://github.com/leonardochappuis/sddmsel/assets/89813338/b0964d44-9361-4e88-91fe-d68035da476b)
![image](https://github.com/leonardochappuis/sddmsel/assets/89813338/7859f6dc-29d2-4cdf-9aed-1076c758b4a6)

lets all love lain

**Comment by leonardochappuis** [comment]
Oof, I will take a look at it today,
Hopefully it's an easy fix.

let's all love lain

**Comment by leonardochappuis** [comment]
It looks like it's something to do with the shaders. My guess is that I used functions from a version of OpenGL that's not supported on your device (or maybe the drivers, as you mentioned that you were having problems with NVIDIA).
I will make a version that loses the CRT-like effect but will probably work on your device.
Later on I will see if I can resolve the issues on the shader.
Meanwhile, you can try removing the ShaderEffect block from "Main.qml" and see if it works for you.

**Comment by WMan22** [comment]
![1](https://github.com/leonardochappuis/sddmsel/assets/48204719/9da33e80-a11f-434c-a25e-2d18346f18b6)
![2](https://github.com/leonardochappuis/sddmsel/assets/48204719/2169ec3c-fcc6-4cae-b676-8f0e8538424b)
![3](https://github.com/leonardochappuis/sddmsel/assets/48204719/96ca256b-3d43-43d1-ab20-8bfea309b3c4)
Having the same problem, vanilla Arch Linux, also nvidia card. I, however, could not hear anything on startup unlike OP.

It might have something to do with the gst-plugin-openh264 dependency, since that does not appear to be in arch's repositories. I am only speculating, however.

**Comment by leonardochappuis** [comment]
You can add it from the AUR @WMan22 
https://aur.archlinux.org/packages/gst-plugin-openh264

In any case, it looks like the problem is with the fragment shader, I'm trying my best to figure out what dependency (if any) is missing or what I can do to make it work "universally".

**Comment by leonardochappuis** [comment]
I have updated the repo guys, please do a git pull.
@pgrwe I have tried to force float conversion in the shaders so that the "sin" function doesn't return errors, but maybe there are errors still.
Let me know if it works!

@WMan22 it seems that you are missing a GStreamer plug-in, plese try installing gst-plugin-openh264 and let me know if it works, if you can!

In any case, I have added a version without shaders to the repo if you guys want to try.

**Comment by pgrwe** [comment]
Removing the ShaderEffect worked - also crazy timing - the updated version with shaders worked like a charm!

**Comment by leonardochappuis** [comment]
Nice, I'm glad! Closing this issue as it seems that the problem was calling the sin function in the fragment shader with uh, "non-float" values.

**Comment by WMan22** [comment]
> @WMan22 it seems that you are missing a GStreamer plug-in, plese try installing gst-plugin-openh264 and let me know if it works, if you can!

~So after installing the plugin from the AUR, the shader now works, but not the video, for some reason. I'm getting the wavy CRT effect now instead of a pink screen, but I'm getting a black background. Now, it's saying it can't view a quicktime decoder, but I have mpv installed, so I'm not sure what the issue is.~
![theme](https://github.com/leonardochappuis/sddmsel/assets/48204719/372affe0-1470-4bfc-bb4e-9ff15221ba30)

**Edit: Never mind, I could have sworn I installed all the pacman dependencies on the readme but I just reinstalled them and it works now, thank you for the help. I didn't know gst-plugin-openh264 was something I had to get from the AUR instead of pacman.**
